### PR TITLE
Implement simple CAST from Map/Tuple/Object to new JSON through serialization/deserialization from JSON string

### PR DESCRIPTION
**Pull request by Avogar**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Implement simple CAST from Map/Tuple/Object to new JSON through serialization/deserialization from JSON string.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by Avogar** [comment]
I don't think we need anything more complex for conversion from `Tuple`/`Map`/`Object('json')` to new `JSON` type and this simple implementation is enough.

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit 9021aeaaff66f7a0c0daeb37d1cd42157c5a15aa with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/71320/9021aeaaff66f7a0c0daeb37d1cd42157c5a15aa/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/71320/9021aeaaff66f7a0c0daeb37d1cd42157c5a15aa/style_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: src/Functions/FunctionsConversion.cpp [hunk 1]
```diff
@@ -4081,23 +4081,43 @@ class FunctionCast final : public IFunctionBase
...
     WrapperType createObjectWrapper(const DataTypePtr & from_type, const DataTypeObject * to_object) const
     {
         if (checkAndGetDataType<DataTypeString>(from_type.get()))
         {
             return [this](ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type, const ColumnNullable * nullable_source, size_t input_rows_count)
             {
-                auto res = ConvertImplGenericFromString<true>::execute(arguments, result_type, nullable_source, input_rows_count, context)->assumeMutable();
-                res->finalize();
-                return res;
+                return ConvertImplGenericFromString<true>::execute(arguments, result_type, nullable_source, input_rows_count, context);
```
Comment (divanik) [inline src/Functions/FunctionsConversion.cpp:4091]: I do not actually understand why you removed finalize here, but if it is ok, I believe you

**Review by divanik — APPROVED** [review]

**Review by divanik — APPROVED** [review]

File: src/Functions/FunctionsConversion.cpp (hunk 1 above)
Comment (Avogar) [reply src/Functions/FunctionsConversion.cpp:4091]: `finalize` makes sense only for columns with old `Object` data type. Here it's unneded and was added only due to bad copy-paste actually

**Pull request merged**